src/gt4py/next/iterator/embedded.py

```diff
@@ -895,7 +935,7 @@ def deref(self) -> Any:
 
         assert self.pos is not None
         shifted_pos = self.pos.copy()
-        axes = _get_axes(self.field)
+        axes = _get_axes(self.field, ignore_zero_dims=True)
```

Comment:
TODO:
1. write iterator tests
2. check if `(vertex_field, vertex_k_field)` should also be valid here.

Reply:
1. I've added two tests.
2. I've added two test named `test_tuple_arg_with_unpromotable_dims` and `test_scalar_arg_with_field`. The implementation is beyond the scope of this PR.